src/Resolution.ts

```diff
@@ -94,18 +101,97 @@ export default class Resolution {
 
   /**
    * Creates a resolution instance with configured provider
-   * @param provider - any provider with sendAsync function impelmented
+   * @param provider - any provider compatable with EIP-1193 (https://eips.ethereum.org/EIPS/eip-1193)
    */
-  static provider(provider: Web3Provider): Resolution {
-    return new this({ blockchain: { web3Provider: provider } });
+  static provider(provider: Provider): Resolution {
```

Comment:
If this method has got its interface changed, I'm in for renaming it to `fromProvider` to make it more consistent with others